test/e2e/query_frontend_test.go

```diff
@@ -717,18 +732,18 @@ func TestRangeQueryDynamicHorizontalSharding(t *testing.T) {
 	))
 
 	// make sure that we don't break cortex cache code.
-	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(3), "cortex_cache_fetched_keys_total"))
+	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(4), "cortex_cache_fetched_keys_total"))
 	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(0), "cortex_cache_hits_total"))
 	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(2), "querier_cache_added_new_total"))
-	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(3), "querier_cache_added_total"))
-	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(3), "querier_cache_misses_total"))
+	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(4), "querier_cache_added_total"))
+	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(4), "querier_cache_misses_total"))
 
 	// Query interval is 2 hours, which is greater than min-slit-interval, query will be broken down into 4 parts
-	// + rest (of interval)
-	testutil.Ok(t, queryFrontend.WaitSumMetrics(e2emon.Equals(5), "thanos_frontend_split_queries_total"))
+	// + rest (of interval) - Doubt - what is rest of interval?
```

Comment:
There is no rest of interval since the specified timestamp is a multiple of `MinQuerySplitInterval` which is one hour.
Let's clean up this.